task/orchestrator_test.go

```diff
@@ -30,10 +33,15 @@ func TestOrchestrator(t *testing.T) {
 	}
 
 	testOnce.Do(func() {
-		// Set the process reap time to 0 to prevent interference between test cases
-		reapTime = 0
+		// Reduce the process reap time to prevent interference between test cases
+		reapTime = 500 * time.Millisecond
 	})
 
+	// Re-parent any child processes to us to simulate the orchestrator being init
+	const PrSetChildSubreaper = 36
+	_, _, err := syscall.Syscall(syscall.SYS_PRCTL, PrSetChildSubreaper, uintptr(1), 0)
+	assert.Check(t, err == 0)
```

Comment:
Discovering this trick took a while, but it let's us ensure the reaper is actually working.